scanner.zig

```diff
@@ -305,7 +305,23 @@ const Message = struct {
                         try writer.writeAll(".{ .");
                         try arg.emitSignature(writer);
                         try writer.writeAll(" = ");
-                        try printIdentifier(writer, arg.name);
+                        if (arg.enum_name != null) {
+                            try writer.writeAll("switch (@typeInfo(");
+                            try arg.emitType(target, writer);
+
+                            // TODO We know the type of the enum at scanning time, but it's
+                            //      currently a bit difficult to access it.
+                            const c_type = if (arg.kind == .uint) "u32" else "i32";
+                            try writer.print(
+                                \\ )) {{
+                                \\    .Enum => @bitCast({}, @enumToInt({})),
```

Comment:
why is this a @bitCast? shouldn't an `@intCast` be sufficient?

Reply:
Yes, that works.

It might have been a left over from when I was trying to just use `@enumToInt`, but some arguments were expecting uint, others int.